src/sentry/static/sentry/app/views/organizationIntegrations/integrationListDirectory.tsx

```diff
@@ -188,22 +190,47 @@ export class OrganizationIntegrations extends AsyncComponent<
   //Returns 0 if uninstalled, 1 if pending, and 2 if installed
   getInstallValue(integration: AppOrProviderOrPlugin) {
     const {integrations} = this.state;
-    if (isSentryApp(integration)) {
-      const install = this.getAppInstall(integration);
-      if (install) {
-        return install.status === 'pending' ? 1 : 2;
-      }
-      return 0;
-    } else if (isPlugin(integration)) {
+
+    if (isPlugin(integration)) {
       return integration.projectList.length > 0 ? 2 : 0;
     }
-    return integrations.find(i => i.provider.key === integration.key) ? 2 : 0;
+
+    if (!isSentryApp(integration)) {
+      return integrations.find(i => i.provider.key === integration.key) ? 2 : 0;
+    }
+
+    const install = this.getAppInstall(integration);
+
+    if (install) {
+      return install.status === 'pending' ? 1 : 2;
+    }
+
+    return 0;
   }
 
+  sortByName = (a: AppOrProviderOrPlugin, b: AppOrProviderOrPlugin) =>
+    a.name.localeCompare(b.name);
+
+  getPopularityWeight = (integration: AppOrProviderOrPlugin) =>
+    POPULARITY_WEIGHT[integration.slug] ?? 1;
```

Comment:
Nit: Feels like `getPopularityWeight` should come before `sortByName`

Reply:
Do you understand the logic for this nit?

Reply:
Just want to check if you can see the logic for this nit?

Reply:
@leedongwei I'm assuming its to put all the `get` functions together and the `sort` functions together, right?

Reply:
Yea but that's not quite the full story. I don't mind if they are mashed together, but it's mostly about consistency. 

You separated out `getInstallValue` earlier and had all the sort functions in their own block. This is fine and you should be consistent about this.

If you're going to keep `getInstallValue` just above `sortByInstalled`. This is fine too. You just have to be consistent and then `getPopularityWeight` should be just above `sortByPopularity`

Reply:
@leedongwei interesting. I understand. 👍